keps/693-multikueue/README.md

```diff
@@ -0,0 +1,234 @@
+We are in talks with SIG-Apps to establish an acceptable way of using a non
+default controller (or none at all)
+
+## Design Details
+MultiKueue will be enabled on a cluster queue using the admission check fields.
+Just like ProvisioningRequest, MultiKueue will have its own configuration, 
+MultiKueueConfig with the following definition. To allow reusing the same clusters
+across many Kueues, additional object, MultiKueueWorkerCluster, is added.
+
+```go
+type MultiKueueConfig struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec MultiKueueConfigSpec `json:"spec,omitempty"`
+}
+
+type MultiKueueConfigSpec struct {
+     // List of MultiKueueWorkerClusters names where the 
+     // workloads from the ClusterQueue should be distributed.
+     Clusters []string `json:"clusters,omitempty"`
+}
+
+type MultiKueueCluster struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec MultiKueueClusterSpec `json:"spec,omitempty"`
+    Status MultiKueueClusterStatus `json:"status,omitempty"`
+}
+
+const (
+    // Location is the path on the disk.
+    PathLocationType LocationType = "Path"
+    
+    // Location is the name of the secret inside kueue-system namespace.
+    SecretLocationType LocationType = "Secret"
+)
+
+type MultiKueueuClusterSpec {
+    // Information how to connect to the cluster.
+    KubeConfig KubeConfig `json:"kubeConfig"`
+}
+
+type KubeConfig struct {
+    // Name of the custer inside the given KubeConfig.
+    Name string `json:"configName"`
+    
+    // Location of the KubeConfig.
+    Location string `json:"location"`
+    
+    // Type of the KubeConfig location.
+    LocationType LocationType `json:"locationType"`
+}
+
+type MultiKueueClusterStatus {
+   Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+```
+
+MultiKueue controller will monitor all cluster definitions and maintain 
+the Kube clients for all of them. Any connectivity problems will be reported both in
+MultiKueuCluster status as well as AdmissionCheckStatus and Events. MultiKueue controller 
+will make sure that whenever the kubeconfig is refreshed, the appropriate 
+clients will also be recreated.
+
+Creation of kubeconfig files is outside of the MultiKueue scope, and is cloud
+provider/environment dependant.
+
+MultiKueue controller, when pushing workloads to the worker clusters, will use the same 
+namespace and local queue names as were used in the management cluster. It is user's 
+responsibility to set up the appropriate namespaces and local queues.
+Worker ClusterQueue definitions may be different than in the management cluster. For example,
+quota settings may be specific to the given location. And/or cluster queue may have different
+admission checks, use ProvisioningRequest, etc.
+
+When distributing the workloads across clusters MultiKueue controller will first create
```

Comment:
What control will MultiKueue have for fair-share?

Reply:
Same as regular, single cluster Kueue - quotas, borrowing, limits, preemptions etc, which can be used at both global and individual cluster level.

Reply:
With this design of MK, any improvements to the fair-share model will directly apply to MK. Resolving for now.